scripts/ci/cache/_utils.mjs

```diff
@@ -0,0 +1,218 @@
+// @ts-check
+import { arch, platform } from 'node:os'
+import crypto from 'node:crypto'
+import { ROOT } from './_common.mjs'
+import { spawn, exec } from 'node:child_process'
+import { resolve, join } from 'node:path'
+import { readFile, readdir, stat } from 'node:fs/promises'
+import { HASH_GENERATE_FILES_SCRIPT } from './_const.mjs'
+
+const GENERATE_HASH_GENERATED_FILES_SCRIPT = resolve(
```

Comment:
could we try/catch/throw on this line?

Reply:
This only returns path, doesn't check if it exists.